apps/researcher/src/lib/community.ts

```diff
@@ -1,24 +1,81 @@
 import {clerkClient, auth} from '@clerk/nextjs';
 import {OrganizationMembership, Organization} from '@clerk/backend/dist/types';
 
-export type Community = Organization;
+export interface Community {
+  id: string;
+  name: string;
+  description?: string;
+  slug: string;
+  imageUrl: string;
+  createdAt: number;
+}
+
+export interface Membership {
+  id: string;
+  role: string;
+  userId: string;
+  firstName: string;
+  lastName: string;
+  imageUrl: string;
+}
+
+function organizationToCommunity(organization: Organization): Community {
+  return {
+    id: organization.id,
+    name: organization.name,
+    // The type of `publicMetadata` is `{ [k: string]: unknown } | null `. Redeclare custom metadata `description`.
+    description: organization.publicMetadata?.description as string | undefined,
+    slug: organization.slug!,
+    imageUrl: organization.imageUrl,
+    createdAt: organization.createdAt,
+  };
+}
+
+function organizationMembershipToCommunityMembership(
+  membership: OrganizationMembership
+): Membership {
+  if (!membership.publicUserData) {
+    throw new Error(
+      'organizationMembershipToCommunityMembership: publicUserData is undefined'
```

Comment:
Should we add the name of the function to the error message? If this error is thrown, the error stack should make clear where the error comes from (e.g. from which function), so adding the name feels redundant?